controllers/apply-next/lib/enrich-form.js

```diff
@@ -0,0 +1,56 @@
+'use strict';
+const { cloneDeep, find, flatMap, reject } = require('lodash');
+const { formatterFor } = require('./formatters');
+
+/**
+ * Enriches a form object
+ *
+ * @param {Object} baseForm
+ * @param {Object} [data]
+ */
+module.exports = function enrichForm(baseForm, data = {}) {
```

Comment:
This method now does a lot less. In fact it only does two things:

1. Assigns values to fields
2. Filters out skippable steps

Even the second part could arguably live in the controller, or be handled in the new richer form model so we might be able to boil this down to a method which is solely concerned with field values.